src/libappimage/core/impl/traversal_type_1.cpp

```diff
@@ -0,0 +1,100 @@
+// system
+#include <fcntl.h>
+#include <cstring>
+#include <iostream>
+
+// library
+#include <archive.h>
+#include <archive_entry.h>
+
+// local
+#include "core/appimage.h"
+#include "core/exceptions.h"
+#include "core/file_istream.h"
+#include "utils/filesystem.h"
+#include "traversal_type_1.h"
+#include "appimage/appimage_shared.h"
+#include "streambuf_fallback.h"
+#include "streambuf_type1.h"
+
+using namespace std;
+using namespace appimage::core::impl;
+
+traversal_type_1::traversal_type_1(const std::string& path) : path(path) {
+    clog << "Opening " << path << " as Type 1 AppImage" << endl;
+
+    a = archive_read_new();
+    archive_read_support_format_iso9660(a);
+    if (archive_read_open_filename(a, path.c_str(), 10240) != ARCHIVE_OK)
+        throw AppImageReadError(archive_error_string(a));
+
+    completed = false;
+}
+
+
+traversal_type_1::traversal_type_1::~traversal_type_1() {
+    clog << "Closing " << path << endl;
+
+    archive_read_close(a);
+    archive_read_free(a);
+}
+
+bool traversal_type_1::isCompleted() {
+    return completed;
+}
+
+std::string traversal_type_1::getEntryName() {
+    if (completed)
+        return std::string();
+
+    const char* entryName = archive_entry_pathname(entry);
+    if (entryName == nullptr)
+        return string();
+
+    // remove ./ prefix from entries names
+    if (strncmp("./", entryName, 2) == 0)
```

Comment:
strncmp can handle it. It will read at most 2 bites, but not necessarily. At this point, we cannot do anything if `entryName` is a malformed string of size 0 without NULL char at the end.

Reply:
Fixed.